src/Deposit.js

```diff
@@ -811,52 +880,75 @@ export default class Deposit {
     }
     /** @type {AutoSubmitState} */
     const state = (this.autoSubmittingState = {})
+    state.fundingTransaction = this.fundingTransaction
+    state.fundingConfirmations = this.fundingConfirmations
 
-    state.fundingTransaction = this.bitcoinAddress.then(async address => {
-      const expectedValue = await this.getSatoshiLotSize()
-
-      console.debug(
-        `Monitoring Bitcoin for transaction to address ${address}...`
-      )
-      return BitcoinHelpers.Transaction.findOrWaitFor(address, expectedValue)
-    })
-
-    state.fundingConfirmations = state.fundingTransaction.then(
-      async transaction => {
-        const requiredConfirmations = parseInt(
-          await this.factory.constantsContract.methods
-            .getTxProofDifficultyFactor()
-            .call()
-        )
-
+    state.proofTransaction = this.state.fundingConfirmations.then(
+      async ({ transaction, requiredConfirmations }) => {
         console.debug(
-          `Waiting for ${requiredConfirmations} confirmations for ` +
+          `Submitting funding proof to deposit ${this.address} for ` +
             `Bitcoin transaction ${transaction.transactionID}...`
         )
-        await BitcoinHelpers.Transaction.waitForConfirmations(
+        const proofArgs = await this.constructFundingProof(
           transaction,
           requiredConfirmations
         )
 
-        return { transaction, requiredConfirmations }
+        return EthereumHelpers.sendSafely(
+          this.contract.methods.provideBTCFundingProof(...proofArgs),
+          {},
+          true
+        )
       }
     )
 
-    state.proofTransaction = state.fundingConfirmations.then(
-      async ({ transaction, requiredConfirmations }) => {
+    return state
+  }
+
+  autoMint() {
+    // Only enable auto-submitting once.
+    if (this.autoMintingState) {
```

Comment:
Should we unify this and `autoSubmitState`? Feels like we shouldn't allow auto-submitting and auto-minting in parallel.

Reply:
Happy to do this. Needs a better name I feel?